tests/test_grouping_utils.py

```diff
@@ -238,10 +239,46 @@ def test_are_identical_pauli_words_non_pauli_word_catch(self, non_pauli_word):
         """Tests TypeError raise for when non-Pauli word Pennylane operators/operations are given
         as input to are_identical_pauli_words."""
 
-        assert pytest.raises(
-            TypeError, are_identical_pauli_words, (non_pauli_word, PauliZ(0) @ PauliZ(1))
-        )
-        assert pytest.raises(
-            TypeError, are_identical_pauli_words, (PauliX("a") @ Identity("b"), non_pauli_word)
-        )
-        assert pytest.raises(TypeError, are_identical_pauli_words, (non_pauli_word, non_pauli_word))
+        with pytest.raises(TypeError):
```

Comment:
I had to change the usage of `pytest.raises` here because it was causing certain lines to not show up in coverage (even though they were being tested)